sdks/python/apache_beam/runners/interactive/interactive_beam.py

```diff
@@ -211,20 +280,27 @@ def show(*pcolls):
         watched_pcollections.add(val)
   for pcoll in pcolls:
     if pcoll not in watched_pcollections:
-      watch({re.sub(r'[\[\]\(\)]', '_', str(pcoll)): pcoll})
+      watch({'anonymous_pcollection_{}'.format(id(pcoll)): pcoll})
 
+  import warnings
+  warnings.filterwarnings('ignore', category=DeprecationWarning)
```

Comment:
This will filter out all deprecation warnings. Not a good outcome, if we would like user to see deprecation warnings.

Reply:
Change the filtering to catch a specific message and only takes effect when `is_in_ipython` when the user invokes `show` for the first time.